{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Connected: root@mysql'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%load_ext sql\n",
    "%config SqlMagic.autocommit=False\n",
    "%sql mysql+pymysql://root:root@127.0.0.1:3306/mysql"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Problem 1: Controls\n",
    "\n",
    "Write a Python script that proves that the lines of data in Germplasm.tsv, and LocusGene are in the same sequence, based on the AGI Locus Code (ATxGxxxxxx). (hint: This will help you decide how to load the data into the database)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Line 1 of data in Germplasm.tsv (AT1G01040) and LocusGene.tsv (AT1G01040) are in the same sequence\n",
      "\n",
      "Line 2 of data in Germplasm.tsv (AT1G01060) and LocusGene.tsv (AT1G01060) are in the same sequence\n",
      "\n",
      "Line 3 of data in Germplasm.tsv (AT1G01140) and LocusGene.tsv (AT1G01140) are in the same sequence\n",
      "\n",
      "Line 4 of data in Germplasm.tsv (AT1G01220) and LocusGene.tsv (AT1G01220) are in the same sequence\n",
      "\n",
      "Line 5 of data in Germplasm.tsv (AT2G03720) and LocusGene.tsv (AT2G03720) are in the same sequence\n",
      "\n",
      "Line 6 of data in Germplasm.tsv (AT2G03800) and LocusGene.tsv (AT2G03800) are in the same sequence\n",
      "\n",
      "Line 7 of data in Germplasm.tsv (AT2G04240) and LocusGene.tsv (AT2G04240) are in the same sequence\n",
      "\n",
      "Line 8 of data in Germplasm.tsv (AT2G05210) and LocusGene.tsv (AT2G05210) are in the same sequence\n",
      "\n",
      "Line 9 of data in Germplasm.tsv (AT3G02130) and LocusGene.tsv (AT3G02130) are in the same sequence\n",
      "\n",
      "Line 10 of data in Germplasm.tsv (AT3G02140) and LocusGene.tsv (AT3G02140) are in the same sequence\n",
      "\n",
      "Line 11 of data in Germplasm.tsv (AT3G02230) and LocusGene.tsv (AT3G02230) are in the same sequence\n",
      "\n",
      "Line 12 of data in Germplasm.tsv (AT3G02260) and LocusGene.tsv (AT3G02260) are in the same sequence\n",
      "\n",
      "Line 13 of data in Germplasm.tsv (AT3G02310) and LocusGene.tsv (AT3G02310) are in the same sequence\n",
      "\n",
      "Line 14 of data in Germplasm.tsv (AT3G02680) and LocusGene.tsv (AT3G02680) are in the same sequence\n",
      "\n",
      "Line 15 of data in Germplasm.tsv (AT3G02850) and LocusGene.tsv (AT3G02850) are in the same sequence\n",
      "\n",
      "Line 16 of data in Germplasm.tsv (AT3G02870) and LocusGene.tsv (AT3G02870) are in the same sequence\n",
      "\n",
      "Line 17 of data in Germplasm.tsv (AT3G03260) and LocusGene.tsv (AT3G03260) are in the same sequence\n",
      "\n",
      "Line 18 of data in Germplasm.tsv (AT4G14790) and LocusGene.tsv (AT4G14790) are in the same sequence\n",
      "\n",
      "Line 19 of data in Germplasm.tsv (AT4G15210) and LocusGene.tsv (AT4G15210) are in the same sequence\n",
      "\n",
      "Line 20 of data in Germplasm.tsv (AT4G15560) and LocusGene.tsv (AT4G15560) are in the same sequence\n",
      "\n",
      "Line 21 of data in Germplasm.tsv (AT4G15570) and LocusGene.tsv (AT4G15570) are in the same sequence\n",
      "\n",
      "Line 22 of data in Germplasm.tsv (AT4G15802) and LocusGene.tsv (AT4G15802) are in the same sequence\n",
      "\n",
      "Line 23 of data in Germplasm.tsv (AT4G15880) and LocusGene.tsv (AT4G15880) are in the same sequence\n",
      "\n",
      "Line 24 of data in Germplasm.tsv (AT4G16420) and LocusGene.tsv (AT4G16420) are in the same sequence\n",
      "\n",
      "Line 25 of data in Germplasm.tsv (AT4G16480) and LocusGene.tsv (AT4G16480) are in the same sequence\n",
      "\n",
      "Line 26 of data in Germplasm.tsv (AT5G10480) and LocusGene.tsv (AT5G10480) are in the same sequence\n",
      "\n",
      "Line 27 of data in Germplasm.tsv (AT5G10510) and LocusGene.tsv (AT5G10510) are in the same sequence\n",
      "\n",
      "Line 28 of data in Germplasm.tsv (AT5G11110) and LocusGene.tsv (AT5G11110) are in the same sequence\n",
      "\n",
      "Line 29 of data in Germplasm.tsv (AT5G11260) and LocusGene.tsv (AT5G11260) are in the same sequence\n",
      "\n",
      "Line 30 of data in Germplasm.tsv (AT5G11510) and LocusGene.tsv (AT5G11510) are in the same sequence\n",
      "\n",
      "Line 31 of data in Germplasm.tsv (AT5G12200) and LocusGene.tsv (AT5G12200) are in the same sequence\n",
      "\n",
      "Line 32 of data in Germplasm.tsv (AT5G13290) and LocusGene.tsv (AT5G13290) are in the same sequence\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import re \n",
    "\n",
    "text1 = open(\"Germplasm.tsv\")\n",
    "text2 = open(\"LocusGene.tsv\")\n",
    "\n",
    "search1 = re.findall(\"AT[1-5]G[0-9]{5}\", text1.read())\n",
    "search2 = re.findall(\"AT[1-5]G[0-9]{5}\", text2.read())\n",
    "\n",
    "#print(search1)\n",
    "#print(search2)\n",
    "\n",
    "\n",
    "count = 0\n",
    "\n",
    "for i in search1:\n",
    "    \n",
    "    if search1[count]==search2[count]:\n",
    "        print(f\"Line {count+1} of data in Germplasm.tsv ({i}) and LocusGene.tsv ({search2[count]}) are in the same sequence\\n\")\n",
    "        count=count+1\n",
    "        \n",
    "    else:\n",
    "        print(f\"Line {count+1} does not match: Germplasm.tsv ({search1[count]}) - LocusGene.tsv ({search2[count]})\\n\")\n",
    "        count=count+1\n",
    "    \n",
    "    \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Problem 2: Design and create the database\n",
    "\n",
    "* It should have two tables - one for each of the two data files.\n",
    "* The two tables should be linked in a 1:1 relationship\n",
    "* you may use either sqlMagic or pymysql to build the database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "6 rows affected.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "    <tr>\n",
       "        <th>Database</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>information_schema</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>Exam2_database</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>germplasm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>mysql</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>performance_schema</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>sys</td>\n",
       "    </tr>\n",
       "</table>"
      ],
      "text/plain": [
       "[('information_schema',),\n",
       " ('Exam2_database',),\n",
       " ('germplasm',),\n",
       " ('mysql',),\n",
       " ('performance_schema',),\n",
       " ('sys',)]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%sql create database Exam2_database;\n",
    "\n",
    "%sql show databases;\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "0 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "0 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "0 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "2 rows affected.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "    <tr>\n",
       "        <th>Tables_in_Exam2_database</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>Germplasm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>LocusGene</td>\n",
       "    </tr>\n",
       "</table>"
      ],
      "text/plain": [
       "[('Germplasm',), ('LocusGene',)]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%sql use Exam2_database;\n",
    "\n",
    "%sql CREATE TABLE Germplasm(id INTEGER NOT NULL AUTO_INCREMENT PRIMARY KEY, Locus VARCHAR(9) NOT NULL, germplasm VARCHAR(400) NOT NULL, phenotype VARCHAR(400) NOT NULL, pubmed INTEGER NOT NULL, LocusGene_id INTEGER NOT NULL);\n",
    "\n",
    "%sql CREATE TABLE LocusGene(id INTEGER NOT NULL AUTO_INCREMENT PRIMARY KEY, Locus VARCHAR(9) NOT NULL, Gene VARCHAR(255) NOT NULL, ProteinLength INTEGER NOT NULL);\n",
    "\n",
    "%sql show tables; \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "0 rows affected.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "    <tr>\n",
       "        <th>id</th>\n",
       "        <th>Locus</th>\n",
       "        <th>germplasm</th>\n",
       "        <th>phenotype</th>\n",
       "        <th>pubmed</th>\n",
       "        <th>LocusGene_id</th>\n",
       "    </tr>\n",
       "</table>"
      ],
      "text/plain": [
       "[]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# In case you want to check it is empty\n",
    "\n",
    "%sql SELECT * FROM Germplasm;\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "0 rows affected.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "    <tr>\n",
       "        <th>id</th>\n",
       "        <th>Locus</th>\n",
       "        <th>Gene</th>\n",
       "        <th>ProteinLength</th>\n",
       "    </tr>\n",
       "</table>"
      ],
      "text/plain": [
       "[]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# In case you want to check it is empty\n",
    "\n",
    "%sql SELECT * FROM LocusGene;\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Problem 3: Fill the database\n",
    "\n",
    "Using pymysql, create a Python script that reads the data from these files, and fills the database. There are a variety of strategies to accomplish this. I will give all strategies equal credit - do whichever one you are most confident with."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n",
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT1G01040\", \"CS3828\", \"Increased abundance of miRNA precursors.\", 17369351, 1)\n",
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT1G01060\", \"lhy-101\", \"The mutant plants are hypersensitive to both FRc and Rc light treatments in hypocotyl elongation and exhibits a small reciprocal enlargement in cotyledon area, albeit not statistically significant.\", 16891401, 2)\n",
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT1G01140\", \"SALK_058629\", \"hypersensitive to low potassium media\", 17486125, 3)\n",
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT1G01220\", \"SALK_012400C\", \"fkgp-1 mutants have about 40 times more L-fucose than wild type Arabidopsis plants, but the levels of other monosaccharides do not appear to differ significantly in the mutants. No obvious phenotypic abnormalities were observed in the fkgp-1 mutants, nor were any differences in the sugar composition of cell wall polysaccharides detected.\", 16367956, 4)\n",
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT2G03720\", \"SALK_042433\", \"Multiple straight hairs\", 16367956, 5)\n",
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT2G03800\", \"gek1-1\", \"Ethanol hypersensitivity.\", 15215505, 6)\n",
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT2G04240\", \"xerico\", \"Resistant to exogenous ABA. Seeds contained lower amounts of endogenous ABA than wildtype.\", 17933900, 7)\n",
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT2G05210\", \"pot1-1\", \"No visible phenotype.\", 17627276, 8)\n",
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT3G02130\", \"rpk2-2\", \"The homozygous progeny is indistinguishable from wild-type plants during vegetative growth but showed several morphological alterations after bolting. These plants displayed enhanced inflorescence branching and formed three times as many siliques and flowers as did wild-type plants.\", 17419837, 9)\n",
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT3G02140\", \"afp4-1\", \"Decreased germination on high concentrations of glucose and sorbitol.\", 18484180, 10)\n",
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT3G02230\", \"rgp1-1\", \"rgp1-1 mutants have significantly lower levels of UDP-L-arabinose mutase activity compared to wild-type plants and significantly lower levels of arabinose in their cell walls.\", 21478444, 11)\n",
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT3G02260\", \"tir3-1 RGLG1:rglg1 rglg2\", \"The triple homozygous progeny has low viability, accumulated anthocyanin, and all plants died before shoot emergence.\", 17586653, 12)\n",
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT3G02310\", \"sep2-1\", \"Non-described subtle phenotype.\", 10821278, 13)\n",
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT3G02680\", \"atnbs1-1\", \"Significantly smaller when grown in the presence of methyl methanosulfonate (MMS) with root growth. Normal growth under standard growth conditions.\", 17672843, 14)\n",
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT3G02850\", \"CS3816\", \"The skor-1 mutant is sensitive to toxic cations in addition to K+ depletion.\", 17568770, 15)\n",
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT3G02870\", \"vtc4-1\", \"ascorbate deficient\", 16595667, 16)\n",
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT3G03260\", \"hdg8-1\", \"No visible phenotype.\", 16778018, 17)\n",
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT4G14790\", \"pdd17\", \"Defective pollen development.\", 19237690, 18)\n",
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT4G15210\", \"bmy1-2\", \"Plants cold-shocked for 6h have an increased starch content compared to wildtype.\", 16297066, 19)\n",
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT4G15560\", \"cla1-1\", \"Mutant seeds grown on medium supplemented with non-phosphorylated synthetic 1-deoxy-D-xylulose (DX) develop green leaves.\", 10982425, 20)\n",
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT4G15570\", \"maa3\", \"Homozygotes are not recovered. Female gametophyte development is delayed and asynchronous. During fertilization, fusion of polar nuclei does not occur. Polar nuclei nucloeli are smaller than WT.\", 18772186, 21)\n",
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT4G15802\", \"Athspb-2\", \"Early flowering, reduced fertility, aborted seeds.\", 20388662, 22)\n",
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT4G15880\", \"esd4-2\", \"Decreased mRNA levels of the floral repressors FLC and MAF4 and increased mRNA levels of the floral activators FT and SOC1.\", 17513499, 23)\n",
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT4G16420\", \"prz1-1\", \"Altered response to auxin and cytokinin\", 12747832, 24)\n",
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT4G16480\", \"atint4-2\", \"No visible phenotype.\", 16603666, 25)\n",
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT5G10480\", \"pas2-3\", \"Segregates 25% embryo lethal.\", 18799749, 26)\n",
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT5G10510\", \"plt3-1\", \"Short roots and shortened root meristem.\", 17960244, 27)\n",
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT5G11110\", \"kns2\", \"Defects are specific to pollen exine structure. Smaller mesh size in the exine structure. Increased number of baculae. Fully fertile.\", 18779216, 28)\n",
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT5G11260\", \"hy5-101\", \"Under FRc conditions, the length mutant hypocotyls is increased compared to that of wild-type plants. Under Rc conditions, the hypocotyl length is also increased and the cotyledon area is smaller.\", 16891401, 29)\n",
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT5G11510\", \"myb3r4-1\", \"No visible phenotype.\", 17287251, 30)\n",
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT5G12200\", \"pyd2-2\", \"The pyd2-2 mutant has a wild-type appearance under normal growth conditions. Pyrimidine nucleotide and uridine levels are not changed in the mutant, but uracil levels are increased. These mutants cannot grow normally when uracil is provided as a sole nitrogen source.\", 19413687, 31)\n",
    "%sql INSERT INTO Germplasm (Locus, germplasm, phenotype, pubmed, LocusGene_id) VALUES (\"AT5G13290\", \"crn-1\", \"Increased meristem size. Vegetative meristems are are 30% larger than wild type. After bolting inflorescence meristems are enlarged and occasionally fasciated. Flowers occasionally produce extra organs in the first 3.\", 12345678, 32)\n",
    "\n",
    "\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT1G01040\", \"DCL1\", 332)\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT1G01060\", \"LHY\", 290)\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT1G01140\", \"CIPK9\", 223)\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT1G01220\", \"FKGP\", 190)\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT2G03720\", \"MRH6\", 189)\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT2G03800\", \"GEK1\", 196)\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT2G04240\", \"XERICO\", 256)\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT2G05210\", \"POT1A\", 221)\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT3G02130\", \"RPK2\", 284)\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT3G02140\", \"TMAC2\", 300)\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT3G02230\", \"RGP1\", 301)\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT3G02260\", \"BIG\",  279)\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT3G02310\", \"SEP2\", 175)\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT3G02680\", \"NBS1\", 190)\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT3G02850\", \"SKOR\", 234)\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT3G02870\", \"VTC4\", 311)\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT3G03260\", \"HDG8\", 194)\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT4G14790\", \"SUV3\", 312)\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT4G15210\", \"BAM5\", 313)\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT4G15560\", \"DXS\", 219)\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT4G15570\", \"MAA3\", 294)\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT4G15802\", \"HSBP\", 254)\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT4G15880\", \"ESD4\", 265)\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT4G16420\", \"ADA2B\", 279)\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT4G16480\", \"INT4\", 284)\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT5G10480\", \"PAS2\", 301)\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT5G10510\", \"AIL6\", 310)\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT5G11110\", \"SPS2\", 232)\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT5G11260\", \"HY5\", 221)\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT5G11510\", \"MYB3R-4\", 336)\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT5G12200\", \"PYD2\", 310)\n",
    "%sql INSERT INTO LocusGene (Locus, Gene, ProteinLength) VALUES (\"AT5G13290\", \"CRN\", 189)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    " # In case you want to check it is filled with data\n",
    "\n",
    "%sql SELECT * FROM Germplasm;\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# In case you want to check it is filled with data\n",
    "\n",
    "%sql SELECT * FROM LocusGene;"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Problem 4: Create reports, written to a file"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**1. Create a report that shows the full, joined, content of the two database tables (including a header line)**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "32 rows affected.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "    <tr>\n",
       "        <th>id</th>\n",
       "        <th>Locus</th>\n",
       "        <th>germplasm</th>\n",
       "        <th>phenotype</th>\n",
       "        <th>pubmed</th>\n",
       "        <th>LocusGene_id</th>\n",
       "        <th>id_1</th>\n",
       "        <th>Locus_1</th>\n",
       "        <th>Gene</th>\n",
       "        <th>ProteinLength</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>1</td>\n",
       "        <td>AT1G01040</td>\n",
       "        <td>CS3828</td>\n",
       "        <td>Increased abundance of miRNA precursors.</td>\n",
       "        <td>17369351</td>\n",
       "        <td>1</td>\n",
       "        <td>1</td>\n",
       "        <td>AT1G01040</td>\n",
       "        <td>DCL1</td>\n",
       "        <td>332</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>2</td>\n",
       "        <td>AT1G01060</td>\n",
       "        <td>lhy-101</td>\n",
       "        <td>The mutant plants are hypersensitive to both FRc and Rc light treatments in hypocotyl elongation and exhibits a small reciprocal enlargement in cotyledon area, albeit not statistically significant.</td>\n",
       "        <td>16891401</td>\n",
       "        <td>2</td>\n",
       "        <td>2</td>\n",
       "        <td>AT1G01060</td>\n",
       "        <td>LHY</td>\n",
       "        <td>290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>3</td>\n",
       "        <td>AT1G01140</td>\n",
       "        <td>SALK_058629</td>\n",
       "        <td>hypersensitive to low potassium media</td>\n",
       "        <td>17486125</td>\n",
       "        <td>3</td>\n",
       "        <td>3</td>\n",
       "        <td>AT1G01140</td>\n",
       "        <td>CIPK9</td>\n",
       "        <td>223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>4</td>\n",
       "        <td>AT1G01220</td>\n",
       "        <td>SALK_012400C</td>\n",
       "        <td>fkgp-1 mutants have about 40 times more L-fucose than wild type Arabidopsis plants, but the levels of other monosaccharides do not appear to differ significantly in the mutants. No obvious phenotypic abnormalities were observed in the fkgp-1 mutants, nor were any differences in the sugar composition of cell wall polysaccharides detected.</td>\n",
       "        <td>16367956</td>\n",
       "        <td>4</td>\n",
       "        <td>4</td>\n",
       "        <td>AT1G01220</td>\n",
       "        <td>FKGP</td>\n",
       "        <td>190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>5</td>\n",
       "        <td>AT2G03720</td>\n",
       "        <td>SALK_042433</td>\n",
       "        <td>Multiple straight hairs</td>\n",
       "        <td>16367956</td>\n",
       "        <td>5</td>\n",
       "        <td>5</td>\n",
       "        <td>AT2G03720</td>\n",
       "        <td>MRH6</td>\n",
       "        <td>189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>6</td>\n",
       "        <td>AT2G03800</td>\n",
       "        <td>gek1-1</td>\n",
       "        <td>Ethanol hypersensitivity.</td>\n",
       "        <td>15215505</td>\n",
       "        <td>6</td>\n",
       "        <td>6</td>\n",
       "        <td>AT2G03800</td>\n",
       "        <td>GEK1</td>\n",
       "        <td>196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>7</td>\n",
       "        <td>AT2G04240</td>\n",
       "        <td>xerico</td>\n",
       "        <td>Resistant to exogenous ABA. Seeds contained lower amounts of endogenous ABA than wildtype.</td>\n",
       "        <td>17933900</td>\n",
       "        <td>7</td>\n",
       "        <td>7</td>\n",
       "        <td>AT2G04240</td>\n",
       "        <td>XERICO</td>\n",
       "        <td>256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>8</td>\n",
       "        <td>AT2G05210</td>\n",
       "        <td>pot1-1</td>\n",
       "        <td>No visible phenotype.</td>\n",
       "        <td>17627276</td>\n",
       "        <td>8</td>\n",
       "        <td>8</td>\n",
       "        <td>AT2G05210</td>\n",
       "        <td>POT1A</td>\n",
       "        <td>221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>9</td>\n",
       "        <td>AT3G02130</td>\n",
       "        <td>rpk2-2</td>\n",
       "        <td>The homozygous progeny is indistinguishable from wild-type plants during vegetative growth but showed several morphological alterations after bolting. These plants displayed enhanced inflorescence branching and formed three times as many siliques and flowers as did wild-type plants.</td>\n",
       "        <td>17419837</td>\n",
       "        <td>9</td>\n",
       "        <td>9</td>\n",
       "        <td>AT3G02130</td>\n",
       "        <td>RPK2</td>\n",
       "        <td>284</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>10</td>\n",
       "        <td>AT3G02140</td>\n",
       "        <td>afp4-1</td>\n",
       "        <td>Decreased germination on high concentrations of glucose and sorbitol.</td>\n",
       "        <td>18484180</td>\n",
       "        <td>10</td>\n",
       "        <td>10</td>\n",
       "        <td>AT3G02140</td>\n",
       "        <td>TMAC2</td>\n",
       "        <td>300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>11</td>\n",
       "        <td>AT3G02230</td>\n",
       "        <td>rgp1-1</td>\n",
       "        <td>rgp1-1 mutants have significantly lower levels of UDP-L-arabinose mutase activity compared to wild-type plants and significantly lower levels of arabinose in their cell walls.</td>\n",
       "        <td>21478444</td>\n",
       "        <td>11</td>\n",
       "        <td>11</td>\n",
       "        <td>AT3G02230</td>\n",
       "        <td>RGP1</td>\n",
       "        <td>301</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>12</td>\n",
       "        <td>AT3G02260</td>\n",
       "        <td>tir3-1 RGLG1:rglg1 rglg2</td>\n",
       "        <td>The triple homozygous progeny has low viability, accumulated anthocyanin, and all plants died before shoot emergence.</td>\n",
       "        <td>17586653</td>\n",
       "        <td>12</td>\n",
       "        <td>12</td>\n",
       "        <td>AT3G02260</td>\n",
       "        <td>BIG</td>\n",
       "        <td>279</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>13</td>\n",
       "        <td>AT3G02310</td>\n",
       "        <td>sep2-1</td>\n",
       "        <td>Non-described subtle phenotype.</td>\n",
       "        <td>10821278</td>\n",
       "        <td>13</td>\n",
       "        <td>13</td>\n",
       "        <td>AT3G02310</td>\n",
       "        <td>SEP2</td>\n",
       "        <td>175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>14</td>\n",
       "        <td>AT3G02680</td>\n",
       "        <td>atnbs1-1</td>\n",
       "        <td>Significantly smaller when grown in the presence of methyl methanosulfonate (MMS) with root growth. Normal growth under standard growth conditions.</td>\n",
       "        <td>17672843</td>\n",
       "        <td>14</td>\n",
       "        <td>14</td>\n",
       "        <td>AT3G02680</td>\n",
       "        <td>NBS1</td>\n",
       "        <td>190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>15</td>\n",
       "        <td>AT3G02850</td>\n",
       "        <td>CS3816</td>\n",
       "        <td>The skor-1 mutant is sensitive to toxic cations in addition to K+ depletion.</td>\n",
       "        <td>17568770</td>\n",
       "        <td>15</td>\n",
       "        <td>15</td>\n",
       "        <td>AT3G02850</td>\n",
       "        <td>SKOR</td>\n",
       "        <td>234</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>16</td>\n",
       "        <td>AT3G02870</td>\n",
       "        <td>vtc4-1</td>\n",
       "        <td>ascorbate deficient</td>\n",
       "        <td>16595667</td>\n",
       "        <td>16</td>\n",
       "        <td>16</td>\n",
       "        <td>AT3G02870</td>\n",
       "        <td>VTC4</td>\n",
       "        <td>311</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>17</td>\n",
       "        <td>AT3G03260</td>\n",
       "        <td>hdg8-1</td>\n",
       "        <td>No visible phenotype.</td>\n",
       "        <td>16778018</td>\n",
       "        <td>17</td>\n",
       "        <td>17</td>\n",
       "        <td>AT3G03260</td>\n",
       "        <td>HDG8</td>\n",
       "        <td>194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>18</td>\n",
       "        <td>AT4G14790</td>\n",
       "        <td>pdd17</td>\n",
       "        <td>Defective pollen development.</td>\n",
       "        <td>19237690</td>\n",
       "        <td>18</td>\n",
       "        <td>18</td>\n",
       "        <td>AT4G14790</td>\n",
       "        <td>SUV3</td>\n",
       "        <td>312</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>19</td>\n",
       "        <td>AT4G15210</td>\n",
       "        <td>bmy1-2</td>\n",
       "        <td>Plants cold-shocked for 6h have an increased starch content compared to wildtype.</td>\n",
       "        <td>16297066</td>\n",
       "        <td>19</td>\n",
       "        <td>19</td>\n",
       "        <td>AT4G15210</td>\n",
       "        <td>BAM5</td>\n",
       "        <td>313</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>20</td>\n",
       "        <td>AT4G15560</td>\n",
       "        <td>cla1-1</td>\n",
       "        <td>Mutant seeds grown on medium supplemented with non-phosphorylated synthetic 1-deoxy-D-xylulose (DX) develop green leaves.</td>\n",
       "        <td>10982425</td>\n",
       "        <td>20</td>\n",
       "        <td>20</td>\n",
       "        <td>AT4G15560</td>\n",
       "        <td>DXS</td>\n",
       "        <td>219</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>21</td>\n",
       "        <td>AT4G15570</td>\n",
       "        <td>maa3</td>\n",
       "        <td>Homozygotes are not recovered. Female gametophyte development is delayed and asynchronous. During fertilization, fusion of polar nuclei does not occur. Polar nuclei nucloeli are smaller than WT.</td>\n",
       "        <td>18772186</td>\n",
       "        <td>21</td>\n",
       "        <td>21</td>\n",
       "        <td>AT4G15570</td>\n",
       "        <td>MAA3</td>\n",
       "        <td>294</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>22</td>\n",
       "        <td>AT4G15802</td>\n",
       "        <td>Athspb-2</td>\n",
       "        <td>Early flowering, reduced fertility, aborted seeds.</td>\n",
       "        <td>20388662</td>\n",
       "        <td>22</td>\n",
       "        <td>22</td>\n",
       "        <td>AT4G15802</td>\n",
       "        <td>HSBP</td>\n",
       "        <td>254</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>23</td>\n",
       "        <td>AT4G15880</td>\n",
       "        <td>esd4-2</td>\n",
       "        <td>Decreased mRNA levels of the floral repressors FLC and MAF4 and increased mRNA levels of the floral activators FT and SOC1.</td>\n",
       "        <td>17513499</td>\n",
       "        <td>23</td>\n",
       "        <td>23</td>\n",
       "        <td>AT4G15880</td>\n",
       "        <td>ESD4</td>\n",
       "        <td>265</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>24</td>\n",
       "        <td>AT4G16420</td>\n",
       "        <td>prz1-1</td>\n",
       "        <td>Altered response to auxin and cytokinin</td>\n",
       "        <td>12747832</td>\n",
       "        <td>24</td>\n",
       "        <td>24</td>\n",
       "        <td>AT4G16420</td>\n",
       "        <td>ADA2B</td>\n",
       "        <td>279</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>25</td>\n",
       "        <td>AT4G16480</td>\n",
       "        <td>atint4-2</td>\n",
       "        <td>No visible phenotype.</td>\n",
       "        <td>16603666</td>\n",
       "        <td>25</td>\n",
       "        <td>25</td>\n",
       "        <td>AT4G16480</td>\n",
       "        <td>INT4</td>\n",
       "        <td>284</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>26</td>\n",
       "        <td>AT5G10480</td>\n",
       "        <td>pas2-3</td>\n",
       "        <td>Segregates 25% embryo lethal.</td>\n",
       "        <td>18799749</td>\n",
       "        <td>26</td>\n",
       "        <td>26</td>\n",
       "        <td>AT5G10480</td>\n",
       "        <td>PAS2</td>\n",
       "        <td>301</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>27</td>\n",
       "        <td>AT5G10510</td>\n",
       "        <td>plt3-1</td>\n",
       "        <td>Short roots and shortened root meristem.</td>\n",
       "        <td>17960244</td>\n",
       "        <td>27</td>\n",
       "        <td>27</td>\n",
       "        <td>AT5G10510</td>\n",
       "        <td>AIL6</td>\n",
       "        <td>310</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>28</td>\n",
       "        <td>AT5G11110</td>\n",
       "        <td>kns2</td>\n",
       "        <td>Defects are specific to pollen exine structure. Smaller mesh size in the exine structure. Increased number of baculae. Fully fertile.</td>\n",
       "        <td>18779216</td>\n",
       "        <td>28</td>\n",
       "        <td>28</td>\n",
       "        <td>AT5G11110</td>\n",
       "        <td>SPS2</td>\n",
       "        <td>232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>29</td>\n",
       "        <td>AT5G11260</td>\n",
       "        <td>hy5-101</td>\n",
       "        <td>Under FRc conditions, the length mutant hypocotyls is increased compared to that of wild-type plants. Under Rc conditions, the hypocotyl length is also increased and the cotyledon area is smaller.</td>\n",
       "        <td>16891401</td>\n",
       "        <td>29</td>\n",
       "        <td>29</td>\n",
       "        <td>AT5G11260</td>\n",
       "        <td>HY5</td>\n",
       "        <td>221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>30</td>\n",
       "        <td>AT5G11510</td>\n",
       "        <td>myb3r4-1</td>\n",
       "        <td>No visible phenotype.</td>\n",
       "        <td>17287251</td>\n",
       "        <td>30</td>\n",
       "        <td>30</td>\n",
       "        <td>AT5G11510</td>\n",
       "        <td>MYB3R-4</td>\n",
       "        <td>336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>31</td>\n",
       "        <td>AT5G12200</td>\n",
       "        <td>pyd2-2</td>\n",
       "        <td>The pyd2-2 mutant has a wild-type appearance under normal growth conditions. Pyrimidine nucleotide and uridine levels are not changed in the mutant, but uracil levels are increased. These mutants cannot grow normally when uracil is provided as a sole nitrogen source.</td>\n",
       "        <td>19413687</td>\n",
       "        <td>31</td>\n",
       "        <td>31</td>\n",
       "        <td>AT5G12200</td>\n",
       "        <td>PYD2</td>\n",
       "        <td>310</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>32</td>\n",
       "        <td>AT5G13290</td>\n",
       "        <td>crn-1</td>\n",
       "        <td>Increased meristem size. Vegetative meristems are are 30% larger than wild type. After bolting inflorescence meristems are enlarged and occasionally fasciated. Flowers occasionally produce extra organs in the first 3.</td>\n",
       "        <td>12345678</td>\n",
       "        <td>32</td>\n",
       "        <td>32</td>\n",
       "        <td>AT5G13290</td>\n",
       "        <td>CRN</td>\n",
       "        <td>189</td>\n",
       "    </tr>\n",
       "</table>"
      ],
      "text/plain": [
       "[(1, 'AT1G01040', 'CS3828', 'Increased abundance of miRNA precursors.', 17369351, 1, 1, 'AT1G01040', 'DCL1', 332),\n",
       " (2, 'AT1G01060', 'lhy-101', 'The mutant plants are hypersensitive to both FRc and Rc light treatments in hypocotyl elongation and exhibits a small reciprocal enlargement in cotyledon area, albeit not statistically significant.', 16891401, 2, 2, 'AT1G01060', 'LHY', 290),\n",
       " (3, 'AT1G01140', 'SALK_058629', 'hypersensitive to low potassium media', 17486125, 3, 3, 'AT1G01140', 'CIPK9', 223),\n",
       " (4, 'AT1G01220', 'SALK_012400C', 'fkgp-1 mutants have about 40 times more L-fucose than wild type Arabidopsis plants, but the levels of other monosaccharides do not appear to differ s ... (41 characters truncated) ... henotypic abnormalities were observed in the fkgp-1 mutants, nor were any differences in the sugar composition of cell wall polysaccharides detected.', 16367956, 4, 4, 'AT1G01220', 'FKGP', 190),\n",
       " (5, 'AT2G03720', 'SALK_042433', 'Multiple straight hairs', 16367956, 5, 5, 'AT2G03720', 'MRH6', 189),\n",
       " (6, 'AT2G03800', 'gek1-1', 'Ethanol hypersensitivity.', 15215505, 6, 6, 'AT2G03800', 'GEK1', 196),\n",
       " (7, 'AT2G04240', 'xerico', 'Resistant to exogenous ABA. Seeds contained lower amounts of endogenous ABA than wildtype.', 17933900, 7, 7, 'AT2G04240', 'XERICO', 256),\n",
       " (8, 'AT2G05210', 'pot1-1', 'No visible phenotype.', 17627276, 8, 8, 'AT2G05210', 'POT1A', 221),\n",
       " (9, 'AT3G02130', 'rpk2-2', 'The homozygous progeny is indistinguishable from wild-type plants during vegetative growth but showed several morphological alterations after bolting. These plants displayed enhanced inflorescence branching and formed three times as many siliques and flowers as did wild-type plants.', 17419837, 9, 9, 'AT3G02130', 'RPK2', 284),\n",
       " (10, 'AT3G02140', 'afp4-1', 'Decreased germination on high concentrations of glucose and sorbitol.', 18484180, 10, 10, 'AT3G02140', 'TMAC2', 300),\n",
       " (11, 'AT3G02230', 'rgp1-1', 'rgp1-1 mutants have significantly lower levels of UDP-L-arabinose mutase activity compared to wild-type plants and significantly lower levels of arabinose in their cell walls.', 21478444, 11, 11, 'AT3G02230', 'RGP1', 301),\n",
       " (12, 'AT3G02260', 'tir3-1 RGLG1:rglg1 rglg2', 'The triple homozygous progeny has low viability, accumulated anthocyanin, and all plants died before shoot emergence.', 17586653, 12, 12, 'AT3G02260', 'BIG', 279),\n",
       " (13, 'AT3G02310', 'sep2-1', 'Non-described subtle phenotype.', 10821278, 13, 13, 'AT3G02310', 'SEP2', 175),\n",
       " (14, 'AT3G02680', 'atnbs1-1', 'Significantly smaller when grown in the presence of methyl methanosulfonate (MMS) with root growth. Normal growth under standard growth conditions.', 17672843, 14, 14, 'AT3G02680', 'NBS1', 190),\n",
       " (15, 'AT3G02850', 'CS3816', 'The skor-1 mutant is sensitive to toxic cations in addition to K+ depletion.', 17568770, 15, 15, 'AT3G02850', 'SKOR', 234),\n",
       " (16, 'AT3G02870', 'vtc4-1', 'ascorbate deficient', 16595667, 16, 16, 'AT3G02870', 'VTC4', 311),\n",
       " (17, 'AT3G03260', 'hdg8-1', 'No visible phenotype.', 16778018, 17, 17, 'AT3G03260', 'HDG8', 194),\n",
       " (18, 'AT4G14790', 'pdd17', 'Defective pollen development.', 19237690, 18, 18, 'AT4G14790', 'SUV3', 312),\n",
       " (19, 'AT4G15210', 'bmy1-2', 'Plants cold-shocked for 6h have an increased starch content compared to wildtype.', 16297066, 19, 19, 'AT4G15210', 'BAM5', 313),\n",
       " (20, 'AT4G15560', 'cla1-1', 'Mutant seeds grown on medium supplemented with non-phosphorylated synthetic 1-deoxy-D-xylulose (DX) develop green leaves.', 10982425, 20, 20, 'AT4G15560', 'DXS', 219),\n",
       " (21, 'AT4G15570', 'maa3', 'Homozygotes are not recovered. Female gametophyte development is delayed and asynchronous. During fertilization, fusion of polar nuclei does not occur. Polar nuclei nucloeli are smaller than WT.', 18772186, 21, 21, 'AT4G15570', 'MAA3', 294),\n",
       " (22, 'AT4G15802', 'Athspb-2', 'Early flowering, reduced fertility, aborted seeds.', 20388662, 22, 22, 'AT4G15802', 'HSBP', 254),\n",
       " (23, 'AT4G15880', 'esd4-2', 'Decreased mRNA levels of the floral repressors FLC and MAF4 and increased mRNA levels of the floral activators FT and SOC1.', 17513499, 23, 23, 'AT4G15880', 'ESD4', 265),\n",
       " (24, 'AT4G16420', 'prz1-1', 'Altered response to auxin and cytokinin', 12747832, 24, 24, 'AT4G16420', 'ADA2B', 279),\n",
       " (25, 'AT4G16480', 'atint4-2', 'No visible phenotype.', 16603666, 25, 25, 'AT4G16480', 'INT4', 284),\n",
       " (26, 'AT5G10480', 'pas2-3', 'Segregates 25% embryo lethal.', 18799749, 26, 26, 'AT5G10480', 'PAS2', 301),\n",
       " (27, 'AT5G10510', 'plt3-1', 'Short roots and shortened root meristem.', 17960244, 27, 27, 'AT5G10510', 'AIL6', 310),\n",
       " (28, 'AT5G11110', 'kns2', 'Defects are specific to pollen exine structure. Smaller mesh size in the exine structure. Increased number of baculae. Fully fertile.', 18779216, 28, 28, 'AT5G11110', 'SPS2', 232),\n",
       " (29, 'AT5G11260', 'hy5-101', 'Under FRc conditions, the length mutant hypocotyls is increased compared to that of wild-type plants. Under Rc conditions, the hypocotyl length is also increased and the cotyledon area is smaller.', 16891401, 29, 29, 'AT5G11260', 'HY5', 221),\n",
       " (30, 'AT5G11510', 'myb3r4-1', 'No visible phenotype.', 17287251, 30, 30, 'AT5G11510', 'MYB3R-4', 336),\n",
       " (31, 'AT5G12200', 'pyd2-2', 'The pyd2-2 mutant has a wild-type appearance under normal growth conditions. Pyrimidine nucleotide and uridine levels are not changed in the mutant, but uracil levels are increased. These mutants cannot grow normally when uracil is provided as a sole nitrogen source.', 19413687, 31, 31, 'AT5G12200', 'PYD2', 310),\n",
       " (32, 'AT5G13290', 'crn-1', 'Increased meristem size. Vegetative meristems are are 30% larger than wild type. After bolting inflorescence meristems are enlarged and occasionally fasciated. Flowers occasionally produce extra organs in the first 3.', 12345678, 32, 32, 'AT5G13290', 'CRN', 189)]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%sql SELECT * FROM Germplasm INNER JOIN LocusGene ON Germplasm.LocusGene_id = LocusGene.id;\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**2. Create a joined report that only includes the Genes SKOR and MAA3**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "2 rows affected.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "    <tr>\n",
       "        <th>id</th>\n",
       "        <th>Locus</th>\n",
       "        <th>germplasm</th>\n",
       "        <th>phenotype</th>\n",
       "        <th>pubmed</th>\n",
       "        <th>LocusGene_id</th>\n",
       "        <th>id_1</th>\n",
       "        <th>Locus_1</th>\n",
       "        <th>Gene</th>\n",
       "        <th>ProteinLength</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>15</td>\n",
       "        <td>AT3G02850</td>\n",
       "        <td>CS3816</td>\n",
       "        <td>The skor-1 mutant is sensitive to toxic cations in addition to K+ depletion.</td>\n",
       "        <td>17568770</td>\n",
       "        <td>15</td>\n",
       "        <td>15</td>\n",
       "        <td>AT3G02850</td>\n",
       "        <td>SKOR</td>\n",
       "        <td>234</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>21</td>\n",
       "        <td>AT4G15570</td>\n",
       "        <td>maa3</td>\n",
       "        <td>Homozygotes are not recovered. Female gametophyte development is delayed and asynchronous. During fertilization, fusion of polar nuclei does not occur. Polar nuclei nucloeli are smaller than WT.</td>\n",
       "        <td>18772186</td>\n",
       "        <td>21</td>\n",
       "        <td>21</td>\n",
       "        <td>AT4G15570</td>\n",
       "        <td>MAA3</td>\n",
       "        <td>294</td>\n",
       "    </tr>\n",
       "</table>"
      ],
      "text/plain": [
       "[(15, 'AT3G02850', 'CS3816', 'The skor-1 mutant is sensitive to toxic cations in addition to K+ depletion.', 17568770, 15, 15, 'AT3G02850', 'SKOR', 234),\n",
       " (21, 'AT4G15570', 'maa3', 'Homozygotes are not recovered. Female gametophyte development is delayed and asynchronous. During fertilization, fusion of polar nuclei does not occur. Polar nuclei nucloeli are smaller than WT.', 18772186, 21, 21, 'AT4G15570', 'MAA3', 294)]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%sql SELECT * FROM Germplasm INNER JOIN LocusGene ON Germplasm.LocusGene_id = LocusGene.id WHERE Gene=\"SKOR\" OR Gene=\"MAA3\";\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**3. Create a report that counts the number of entries for each Chromosome (AT1Gxxxxxx to AT5Gxxxxxxx)**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "    <tr>\n",
       "        <th>AT1 matches</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>4</td>\n",
       "    </tr>\n",
       "</table>"
      ],
      "text/plain": [
       "[(4,)]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Counts for chromosome 1 (they all begin with AT1)\n",
    "\n",
    "%sql SELECT COUNT(Locus) AS \"AT1 matches\" FROM LocusGene WHERE Locus LIKE \"AT1%\";\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "    <tr>\n",
       "        <th>AT2 matches</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>4</td>\n",
       "    </tr>\n",
       "</table>"
      ],
      "text/plain": [
       "[(4,)]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Counts for chromosome 2 (they all begin with AT2)\n",
    "\n",
    "%sql SELECT COUNT(Locus) AS \"AT2 matches\" FROM LocusGene WHERE Locus LIKE \"AT2%\";"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "    <tr>\n",
       "        <th>AT3 matches</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>9</td>\n",
       "    </tr>\n",
       "</table>"
      ],
      "text/plain": [
       "[(9,)]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Counts for chromosome 3 (they all begin with AT3)\n",
    "\n",
    "%sql SELECT COUNT(Locus) AS \"AT3 matches\" FROM LocusGene WHERE Locus LIKE \"AT3%\";"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "    <tr>\n",
       "        <th>AT4 matches</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>8</td>\n",
       "    </tr>\n",
       "</table>"
      ],
      "text/plain": [
       "[(8,)]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Counts for chromosome 4 (they all begin with AT4)\n",
    "\n",
    "%sql SELECT COUNT(Locus) AS \"AT4 matches\" FROM LocusGene WHERE Locus LIKE \"AT4%\";"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "    <tr>\n",
       "        <th>AT5 matches</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>7</td>\n",
       "    </tr>\n",
       "</table>"
      ],
      "text/plain": [
       "[(7,)]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Counts for chromosome 5 (they all begin with AT5)\n",
    "\n",
    "%sql SELECT COUNT(Locus) AS \"AT5 matches\" FROM LocusGene WHERE Locus LIKE \"AT5%\";"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**4. Create a report that shows the average protein length for the genes on each Chromosome (AT1Gxxxxxx to AT5Gxxxxxxx)**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "    <tr>\n",
       "        <th>avg protein length on AT1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>258.7500</td>\n",
       "    </tr>\n",
       "</table>"
      ],
      "text/plain": [
       "[(Decimal('258.7500'),)]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Protein length average on chromosomes 1\n",
    "\n",
    "%sql SELECT AVG(ProteinLength) AS \"avg protein length on AT1\" FROM LocusGene WHERE Locus LIKE \"AT1%\";\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "    <tr>\n",
       "        <th>avg protein length on AT2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>215.5000</td>\n",
       "    </tr>\n",
       "</table>"
      ],
      "text/plain": [
       "[(Decimal('215.5000'),)]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Protein length average on chromosomes 2\n",
    "\n",
    "%sql SELECT AVG(ProteinLength) AS \"avg protein length on AT2\" FROM LocusGene WHERE Locus LIKE \"AT2%\";\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "    <tr>\n",
       "        <th>avg protein length on AT3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>252.0000</td>\n",
       "    </tr>\n",
       "</table>"
      ],
      "text/plain": [
       "[(Decimal('252.0000'),)]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Protein length average on chromosomes 3\n",
    "\n",
    "%sql SELECT AVG(ProteinLength) AS \"avg protein length on AT3\" FROM LocusGene WHERE Locus LIKE \"AT3%\";\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "    <tr>\n",
       "        <th>avg protein length on AT4</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>277.5000</td>\n",
       "    </tr>\n",
       "</table>"
      ],
      "text/plain": [
       "[(Decimal('277.5000'),)]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Protein length average on chromosomes 4\n",
    "\n",
    "%sql SELECT AVG(ProteinLength) AS \"avg protein length on AT4\" FROM LocusGene WHERE Locus LIKE \"AT4%\";\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " * mysql+pymysql://root:***@127.0.0.1:3306/mysql\n",
      "1 rows affected.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "    <tr>\n",
       "        <th>avg protein length on AT5</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>271.2857</td>\n",
       "    </tr>\n",
       "</table>"
      ],
      "text/plain": [
       "[(Decimal('271.2857'),)]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Protein length average on chromosomes 5\n",
    "\n",
    "%sql SELECT AVG(ProteinLength) AS \"avg protein length on AT5\" FROM LocusGene WHERE Locus LIKE \"AT5%\";\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<pre>\n",
    "\n",
    "\n",
    "\n",
    "<pre>\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# In case you wanted to remove the database created in the exam, run this command:\n",
    "\n",
    "%sql drop database Exam2_database;"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
